ActivatedPlateletSource.cpp

```diff
@@ -0,0 +1,171 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2014 - 2020 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+#include <ibamr/config.h>
+
+#include <ibamr/app_namespaces.h>
+
+#include <HierarchyDataOpsManager.h>
+#include <SAMRAI_config.h>
+#include <math.h>
+
+// Local includes
+#include "ActivatedPlateletSource.h"
+
+/////////////////////////////// PUBLIC ///////////////////////////////////////
+
+ActivatedPlateletSource::ActivatedPlateletSource(Pointer<Variable<NDIM>> pl_n_var,
+                                                 Pointer<Variable<NDIM>> c_var,
+                                                 Pointer<AdvDiffHierarchyIntegrator> adv_diff_hier_integrator)
+    : d_pl_n_var(pl_n_var), d_c_var(c_var), d_adv_diff_hier_integrator(adv_diff_hier_integrator)
+{
+    // These need to be changed to the relevant parameters
+    // a0 Constants
+    d_Kua = input_db->detDouble("Kua");
+    d_Kuw = input_db->getDouble("Kuw");
+    // w constant
+    d_w_mx = input_db->getDouble("wmax");
+    return;
+} // ActivatedPlateletSource
+
+bool
+ActivatedPlateletSource::isTimeDependent() const
+{
+    return true;
+} // isTimeDependent
+
+void
+ActivatedPlateletSource::setDataOnPatchHierarchy(const int data_idx,
+                                                 Pointer<Variable<NDIM>> var,
+                                                 Pointer<PatchHierarchy<NDIM>> hierarchy,
+                                                 const double data_time,
+                                                 const bool initial_time,
+                                                 const int coarsest_ln_in,
+                                                 const int finest_ln_in)
+{
+    // Loop over variables.
+    std::array<Pointer<Variable<NDIM>>, 2> vars = { d_pl_n_var, d_c_var };
```

Comment:
This needs to be the `Variable` that are used in `setDataOnPatch`.
```suggestion
    std::array<Pointer<Variable<NDIM>>, 2> vars = { d_pl_n_var, d_w_var, d_pl_a_var };
```